cmd/gaia/app/test_utils.go

```diff
@@ -0,0 +1,76 @@
+package app
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/gov"
+	"github.com/cosmos/cosmos-sdk/x/stake"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+// NewTestGaiaAppGenState creates the core parameters for a test genesis
+// initialization given a set of genesis txs, TM validators and their respective
+// operating addresses.
+func NewTestGaiaAppGenState(
+	cdc *codec.Codec, appGenTxs []json.RawMessage, tmVals []tmtypes.GenesisValidator, valOperAddrs []sdk.ValAddress,
+) (GenesisState, error) {
+
+	switch {
+	case len(appGenTxs) == 0:
+		return GenesisState{}, errors.New("must provide at least genesis transaction")
+	case len(tmVals) != len(valOperAddrs):
+		return GenesisState{}, errors.New("number of TM validators does not match number of operator addresses")
+	}
+
+	// start with the default staking genesis state
+	stakeData := stake.DefaultGenesisState()
+
+	// get genesis account information
+	genAccs := make([]GenesisAccount, len(appGenTxs))
+	for i, appGenTx := range appGenTxs {
+
+		var genTx GaiaGenTx
+		if err := cdc.UnmarshalJSON(appGenTx, &genTx); err != nil {
+			return GenesisState{}, err
+		}
+
+		stakeData.Pool.LooseTokens = stakeData.Pool.LooseTokens.Add(sdk.NewDecFromInt(freeFermionsAcc))
```

Comment:
see comment above. I'm also curious why is it called `fermions` and not `atoms` for example?

Reply:
This was taking from the existing non-test app gen state function. It just allocated some tokens to the pool. Nothing to do with power.

Reply:
ok, nbd